frontend/src/lib/lemon-ui/LemonSelect/LemonSearchableSelect.tsx

```diff
@@ -0,0 +1,124 @@
+import Fuse from 'fuse.js'
+import { useMemo, useState } from 'react'
+
+import { LemonInput } from '@posthog/lemon-ui'
+
+import {
+    LemonSelect,
+    LemonSelectOption,
+    LemonSelectOptions,
+    LemonSelectPropsBase,
+    LemonSelectPropsClearable,
+    LemonSelectPropsNonClearable,
+    LemonSelectSection,
+    isLemonSelectSection,
+} from './LemonSelect'
+
+export interface LemonSearchableSelectPropsBase<T> extends LemonSelectPropsBase<T> {
+    searchPlaceholder?: string
+}
+
+export interface LemonSearchableSelectPropsClearable<T>
+    extends LemonSearchableSelectPropsBase<T>,
+        LemonSelectPropsClearable<T> {}
+
+export interface LemonSearchableSelectPropsNonClearable<T>
+    extends LemonSearchableSelectPropsBase<T>,
+        LemonSelectPropsNonClearable<T> {}
+
+export type LemonSearchableSelectProps<T> =
+    | LemonSearchableSelectPropsClearable<T>
+    | LemonSearchableSelectPropsNonClearable<T>
+
+function flattenOptions<T>(options: LemonSelectOptions<T>): LemonSelectOption<T>[] {
```

Comment:
Why do we need these functions here but we didn't need it for `LemonSelect`? Is it just because of the filtering behavior?